runtime/drivers/clickhouse/clickhouse.go

```diff
@@ -447,14 +446,32 @@ func (c *Connection) Config() map[string]any {
 func (c *Connection) Close() error {
 	c.cancel()
 
-	errDB := c.db.Close()
+	var errors []error
+
+	if c.readDB != nil {
+		if err := c.readDB.Close(); err != nil {
+			errors = append(errors, fmt.Errorf("closing read connection: %w", err))
+		}
+	}
+
+	if c.writeDB != nil && c.writeDB != c.readDB {
+		if err := c.writeDB.Close(); err != nil {
+			errors = append(errors, fmt.Errorf("closing write connection: %w", err))
+		}
+	}
 
 	var errEmbed error
 	if c.embed != nil {
 		errEmbed = c.embed.stop()
+		if errEmbed != nil {
+			errors = append(errors, fmt.Errorf("stopping embedded ClickHouse: %w", errEmbed))
+		}
 	}
 
-	return errors.Join(errDB, errEmbed)
+	if len(errors) > 0 {
+		return fmt.Errorf("multiple errors during close: %v", errors)
+	}
+	return nil
```

Comment:
Why this change? Using `%v` to list the errors breaks the ability to use `errors.Is` to handle errors (e.g. context cancellations).